internal/stubserver/stubserver.go

```diff
@@ -127,24 +127,58 @@ func (ss *StubServer) StartServer(sopts ...grpc.ServerOption) error {
 		var err error
 		lis, err = net.Listen(ss.Network, ss.Address)
 		if err != nil {
-			return fmt.Errorf("net.Listen(%q, %q) = %v", ss.Network, ss.Address, err)
+			return nil, fmt.Errorf("net.Listen(%q, %q) = %v", ss.Network, ss.Address, err)
 		}
 	}
 	ss.Address = lis.Addr().String()
-	ss.cleanups = append(ss.cleanups, func() { lis.Close() })
 
-	s := grpc.NewServer(sopts...)
+	ss.S = grpc.NewServer(sopts...)
 	for _, so := range sopts {
 		switch x := so.(type) {
 		case *registerServiceServerOption:
-			x.f(s)
+			x.f(ss.S)
+		}
+	}
+
+	testgrpc.RegisterTestServiceServer(ss.S, ss)
+	ss.cleanups = append(ss.cleanups, ss.S.Stop, func() { lis.Close() })
```

Comment:
Minor detail: Since `grpc.Server.Stop` calls `lis.Close()`, is it really required here?

Reply:
True...it's a bit weird since the server isn't even serving on the listener in the case of the handler server.  But yeah we don't need this.

Reply:
Oh.. wait in the handler server case we never pass the listener to grpc..so we still need the close. But only on the handler path.  Added there.